kubernetes/metadata/resource_test.go

```diff
@@ -262,3 +262,102 @@ func TestNamespaceAwareResource_GenerateWithNamespace(t *testing.T) {
 		})
 	}
 }
+
+func Test_generateMapSubset(t *testing.T) {
+
+	Labelsdedot := map[string]string{
+		"app.kubernetes.io/name": "no",
+		"foo":                    "bar",
+	}
+
+	Labels := map[string]string{
+		"nottomatch":       "no",
+		"foo":              "bar",
+		"foo1":             "bar1",
+		"foo2":             "bar2",
+		"foo-example":      "bar-example",
+		"test":             "test1",
+		"footest":          "footest1",
+		"simplefo-example": "simplefo-example1",
+	}
+	//Validating wildcards
+	key1 := []string{
+		"foo*",
+		"test",
+	}
+	expectedresult1 := mapstr.M{
+		"foo":              "bar",
+		"foo1":             "bar1",
+		"foo2":             "bar2",
+		"foo-example":      "bar-example",
+		"test":             "test1",
+		"footest":          "footest1",
+		"simplefo-example": "simplefo-example1",
+	}
+
+	output1 := generateMapSubset(Labels, key1, false, true)
+
+	//Validating ?
+	key2 := []string{
+		"0?1",
+	}
+	expectedresult2 := mapstr.M{
+		"foo1": "bar1",
+	}
+	output2 := generateMapSubset(Labels, key2, false, true)
+
+	//Validating start of a string
+	key3 := []string{
+		"^test",
+	}
+	expectedresult3 := mapstr.M{
+		"test": "test1",
+	}
+	output3 := generateMapSubset(Labels, key3, false, true)
+
+	//Validating end of a string
+	key4 := []string{
+		"test$",
+	}
+	expectedresult4 := mapstr.M{
+		"test":    "test1",
+		"footest": "footest1",
+	}
+	output4 := generateMapSubset(Labels, key4, false, true)
+
+	//Exact matches
+	key5 := []string{
+		"t{2}",
+	}
+	expectedresult5 := mapstr.M{
+		"nottomatch": "no",
+	}
+	output5 := generateMapSubset(Labels, key5, false, true)
+
+	assert.Equal(t, expectedresult1, output1)
```

Comment:
nit: Could use a for loop pattern like what happens in other tests as well: https://github.com/elastic/elastic-agent-autodiscover/blob/005a08bd7cb98db6051fb8c7532ade98d258b892/kubernetes/metadata/resource_test.go#L131-L134

Reply:
Tests Updated @ChrsMark !

```bash
go test -timeout 30s -run ^Test_generateMapSubset$ github.com/elastic/elastic-agent-autodiscover/kubernetes/metadata

ok  	github.com/elastic/elastic-agent-autodiscover/kubernetes/metadata	1.944s
```